edx_sga/tasks.py

```diff
@@ -0,0 +1,309 @@
+        item_id=block_id
+    )
+    for submissions_student in submissions_students:
+        submissions = _get_submissions(submissions_student.student_id, course_id, block_id)
+        if submissions:
+            answer = submissions[0]['answer']
+            if answer:
+                student = user_by_anonymous_id(submissions_student.student_id)
+                source_file = six.u('{loc.org}/{loc.course}/{loc.block_type}/{loc.block_id}/{sha1}{ext}').format(
+                    loc=location,
+                    sha1=answer['sha1'],
+                    ext=os.path.splitext(answer['filename'])[1]
+                )
+
+                destination_file = os.path.join(
+                    destination_path,
+                    "{student_name}_{sha1}{ext}".format(
+                        student_name=student.username,
+                        sha1=answer['sha1'],
+                        ext=os.path.splitext(answer['filename'])[1]
+                    )
+                )
+                try:
+                    file_descriptor = default_storage.open(source_file)
+                    with open(destination_file, 'wb') as file_pointer:
+                        while True:
+                            data = file_descriptor.read(BLOCK_SIZE)
+                            if data == '':  # end of file reached
+                                break
+                            file_pointer.write(data)
+
+                except IOError:
+                    log.exception("Unable to download submission of student=%s", student.username)
+        else:
+            continue
+
+
+def _compress_folder(destination_path):
+    """
+    compress given folder/file
+
+    Args:
+        destination_path (str): path (including name) of folder/file which we want to compress.
+    """
+    make_archive(
+        destination_path,
+        'zip',
+        destination_path
+    )
+
+
+@CELERY_APP.task
+def zip_student_submissions(course_id, block_id, location, user_name):
+    """
+    Task to download all submissions as zip file
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        location (Location): location of sga module
+        user_name (str): staff user name
+    """
+    submissions_dir_name = get_submissions_dir_name(user_name, block_id, course_id)
+    destination_path = _get_dir_path(submissions_dir_name, location)
+    _mk_dir(destination_path)
+    remove_zip_file(user_name, block_id, course_id)
+    _compress_student_submissions(block_id, course_id, location, destination_path)
+    _compress_folder(destination_path)
+
+
+def get_submissions_dir_name(user_name, block_id, course_id):
+    """
+    returns submission folder name where we are saving files temporary.
+
+    Args:
+        course_id (unicode): edx course id
+        block_id (unicode): edx block id
+        user_name (unicode): staff user name
+    """
+    return "{username}_submissions_{id}_{course_key}".format(
```

Comment:
course key is required as part of zip file, please see #184.
username is also required
i added block id because there can be more the 1 SGA in a course.

cc @pdpinch